### ISSUE TITLE
How can I add masks to parameters

### ISSUE THREAD

**Issue by tzm1003306213** [pytorch/pytorch#30962]
Hi,

Can I use hook to add a parameter masking function to Conv2d. Specifically, I’d like to add a binary mask buffer to each Conv2d module, during each training step, I need to update the mask buffer and then use it to mask the weight.

Or, is there any method to add masks and apply the masks to Conv2d in a given model.

Thanks!

**Comment by SebastienEske** [comment]
Hello,

I am not an expert but one (costly) way to do it would be with a copy of the network weights
- 1 original network net_orig
- 1 mask network net_mask
- 1 computation network net_calc, with shared weights with net_orig except for the Conv2d

then you can do:
```
copy net_orig*net_mask into net_calc Conv2d modules
forward net_calc
backward net_calc
apply solver on net_calc
if deep copy (shouldn't need that), copy back non-masked weights from net_calc into net_orig
update net_mask
```
Hope this helps. There are a few masking features for tensors already here https://pytorch.org/docs/stable/tensors.html They should help with the copying.

Other option is to implement the masked convolution yourself. :-)
